src/server.ts

```diff
@@ -197,12 +261,12 @@ class Server extends EventEmitter {
      * @param socket
      * @param head The first packet of the tunneling stream (may be empty)
      */
-    async onConnect(request, socket, head) {
+    async onConnect(request: http.IncomingMessage, socket: Socket, head: Buffer): Promise<void> {
         try {
             const handlerOpts = await this.prepareRequestHandling(request);
             handlerOpts.srcHead = head;
 
-            const data = { request, sourceSocket: socket, head, handlerOpts, server: this };
+            const data = { request, sourceSocket: socket, head, handlerOpts: handlerOpts as ChainOpts, server: this, isPlain: false };
```

Comment:
we could move the type assertion to `handlerOpts` definition (line 266)

Reply:
We actually can't do this because it's different in case of `direct` handler.